commands/check.go

```diff
@@ -49,7 +49,8 @@ func (c *Check) Execute() error {
 		return fmt.Errorf("invalid payload: %s", err)
 	}
 
-	if req.Source.AwsAccessKeyId != "" && req.Source.AwsSecretAccessKey != "" && req.Source.AwsRegion != "" {
+	if (req.Source.AwsAccessKeyId != "" && req.Source.AwsSecretAccessKey != "" && req.Source.AwsRegion != "") ||
+	    ( req.Source.AwsEC2Credentials ){
```

Comment:
It looks like `source.AwsRegion` is required for both auth methods:

```suggestion
	    ((req.Source.AwsAccessKeyId != "" && req.Source.AwsSecretAccessKey != "") || (req.Source.AwsEC2Credentials != "")) && source.AwsRegion != "" {
```

Reply:
Good find, I will add that.